src/main/scala/com/databricks/spark/xml/parsers/StaxXmlGenerator.scala

```diff
@@ -40,7 +40,8 @@ private[xml] object StaxXmlGenerator {
       (vt, v) match {
         // If this is meant to be attribute, write an attribute
         case (_, null) | (NullType, _) if name.startsWith(options.attributePrefix) =>
-          writer.writeAttribute(name.substring(options.attributePrefix.size), options.nullValue)
+        // Because usually attributes having `null` do not exist in elements, just do not write
+        // attributes when given values are `null`.
```

Comment:
I wonder if not writing attributes for null is appropriate. I think I should make a PR separately for this and the bug for reading attributes.